_posts/2022-01-14-immutables-library.md

```diff
@@ -0,0 +1,793 @@
+
+# Styles
+The `@Value.Style` is the annotation with which we control what will the annotation process generate. In the [prevous example](#using-the-new-method), we used the `@Value.Style` annotation to generate the standard constructor format. 
+
+We can use the annotation on several levels:
+- on the package level
+- on the top class level
+- on the nested class level
+- on the annotation level
+
+The package level annotation will apply the used style to the whole package. 
+
+The class level will take effect on the class where we placed it and on all nested classes. 
+
+The annotation level class will create the new annotation as the meta-annotation. We can then use this new meta-annotation for class levels.
+
+__Note: It is the recommendation to use one or more meta-annotations instead of class or package level__
+
+There are several things that we need to be aware of:
+- If there is mixing in the applied styles they will be selected indeterministic. They are never merged.
+- Style can be a powerful tool, and we need to be careful when using them.
+- Styles are cached. When changing something on the style, we need to rebuild the project or even restart the IDE.
+
+## Style Meta-annotation Definition
+Let us look how to define new meta-annotation:
+```java
+@Target({ElementType.PACKAGE,ElementType.TYPE})
+@Retention(RetentionPolicy.CLASS)
+@Value.Style(
+        of = "new",
+        strictBuilder = true,
+        allParameters = true,
+        visibility = Value.Style.ImplementationVisibility.PUBLIC
+
+)
+public @interface CustomStyle {}
+```
+After defining `@Target` and `@Retention` annotation, we come to the `@Value.Style` annotation. The first value defined that we want to use the `new` constructor. The next thing that we define is that we want to use the strictBuilder and that all attributes should be annotated with the `@Value.Parameter` annotation. The last style defined is that the implementation visibility will be public.
+
+For more information about style possibilities, please refer to the [official documentation.](https://immutables.github.io/style.html)
+
+# Conclusion
+We saw how the immutable library helps us build immutable, thread-safe, and null-safe domain objects. It helps us build clean and readable POJO classes. 
+
+Since it is a powerful tool we need to be careful how to use it. We can easily stray down the wrong path and overuse its features.
+
+The last thing that I want to point out is the `@Value.Style` annotation. The `@Value.Immutable` annotation tells what will be generated, while the `@Value.Style` tells how will be generated. This annotation can be a slippery slope, and we need to be careful and go outside of the default setting only when we are certain that we need to.
+
+For deeper reading on the immutable library please refer to the [official page.](https://immutables.github.io/)
```

Comment:
Add a link to your code examples again as a last sentence so readers can look up the example code there.